### PR TITLE
fix(protocol-designer): fix hotkey text based on PC or Mac

### PR DESCRIPTION
**Pull request by syao1226**
fix RQA-3668

# Overview

updating the `ProtocolStep` to display hotkey based on PC or Mac

### CODE REVIEW

**Review by jerader — APPROVED** [review]
works well on mac! maybe we can see if Alex or someone can test this quickly on a windows.

**Comment by koji** [comment]
I think it’s a good idea to use a custom hook.

```ts
import { useState, useEffect } from "react";

export const useOS = () => {
  const [os, setOS] = useState<string | null>(null);

  useEffect(() => {
    const userAgent = window.navigator.userAgent;

    if (userAgent.includes("Win")) {
      setOS("Windows");
    } else if (userAgent.includes("Mac")) {
      setOS("MacOS");
    } else if (userAgent.includes("Linux")) {
      setOS("Linux");
    } else {
      // smartphones, tablets and others
      setOS("Unknown");
    }
  }, []);

  return os;
};
```

I tested the above code with mac, Windows, and Ubuntu.

File: protocol-designer/src/pages/Designer/ProtocolSteps/index.tsx [hunk 1]
```diff
@@ -62,6 +62,15 @@ export function ProtocolSteps(): JSX.Element {
...
   const stepDetails = currentStep?.stepDetails ?? null
+
+  let commandText = t('command_click_to_multi_select')
+
+  if (navigator.userAgent.includes('Mac')) {
+    commandText = t('mac_command_click_to_multi_select')
+  } else if (navigator.userAgent.includes('Windows')) {
+    commandText = t('windows_command_click_to_multi_select')
+  }
+
```
Comment (koji) [inline protocol-designer/src/pages/Designer/ProtocolSteps/index.tsx]:
```suggestion
  const os = useOS()
  let commandText = t('command_click_to_multi_select')
  if(os === 'MacOS') {
    commandText = t('mac_command_click_to_multi_select')
  }
  if(os === 'Windows') {
    commandText = t('windows_command_click_to_multi_select')
  }
```

File: protocol-designer/src/resources/useOS.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { useState, useEffect } from "react";
+
+export const useOS = () => {
```
Comment (koji) [inline protocol-designer/src/resources/useOS.ts]:
```suggestion
export const useOS = (): string => {
```

**Pull request closed without merging**